### PR TITLE
Online DDL: `ANALYZE` the shadow table before cut-over

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

See story in https://github.com/vitessio/vitess/issues/17200

Online DDL will now run `ANALYZE TABLE` on the shadow table before cut-over. Note:
- We only run `ANALYZE` once, on the first cut-over. Successive cut-overs (in case of continued timeouts) will not run `ANALYZE TABLE`.
- We use the cut-over threshold as `lock_wait_timeout` to prevent excessive locking.
- `ANALYZE` is normally fast, but if runs long, it could cause vreplication lag. We therefore run this before locking the tables, and just before creating the sentry table, after which we issue `waitForPos`.
- `ANALYZE` is best effort. If it fails (timeout, likely) we take note but proceed with cut-over.

## Related Issue(s)

Fixes https://github.com/vitessio/vitess/issues/17200

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [ ] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vttablet/onlineddl/executor.go [hunk 1]
```diff
@@ -967,13 +999,13 @@ func (e *Executor) cutOverVReplMigration(ctx context.Context, s *VReplStream, sh
 	if err != nil {
 		return vterrors.Wrapf(err, "failed getting locking connection")
 	}
+	defer lockConn.Recycle()
```
Comment (shlomi-noach) [inline go/vt/vttablet/onlineddl/executor.go:1002]: This is an unrelated change that I spotted, and will also make a separate PR just for this change.

**Comment by GrahamCampbell** [comment]
This is a great idea. We've sometimes had to run this manually after cut-over to correct query plans.

File: go/vt/vttablet/onlineddl/executor.go [hunk 2]
```diff
@@ -942,12 +943,43 @@ func (e *Executor) cutOverVReplMigration(ctx context.Context, s *VReplStream, sh
...
-		parsed := sqlparser.BuildParsedQuery(sqlCreateSentryTable, sentryTableName)
-		if _, err := e.execQuery(ctx, parsed.Query); err != nil {
-			return vterrors.Wrapf(err, "failed creating sentry table")
-		}
-		e.updateMigrationStage(ctx, onlineDDL.UUID, "sentry table created: %s", sentryTableName)
 
+		preparation := func() error {
+			preparationsConn, err := e.pool.Get(ctx, nil)
+			if err != nil {
+				return vterrors.Wrapf(err, "failed getting preparation connection")
```
Comment (mattlord) [inline go/vt/vttablet/onlineddl/executor.go:950]: This should just be Wrap. The linters are starting to get more strict about this. Same thing for the other instances of Wrapf w/o any format options/params. I know it's nitty, but otherwise people with newer linters locally will encounter the warnings/errors and have to fix them (e.g. simply when merging in the latest from `main`).

**Review by mattlord — APPROVED** [review]
LGTM! Had the one Wrapf nit and then the larger question about the naming, since I think `shadow` would be better than `vrepl` since I didn't see anything vreplication specific. No?

File: go/vt/sidecardb/schema/onlineddl/schema_migrations.sql [hunk 3]
```diff
@@ -72,6 +72,7 @@ CREATE TABLE IF NOT EXISTS schema_migrations
     `is_immediate_operation`          tinyint unsigned NOT NULL DEFAULT '0',
     `reviewed_timestamp`              timestamp        NULL DEFAULT NULL,
     `ready_to_complete_timestamp`     timestamp        NULL DEFAULT NULL,
+    `vrepl_analyzed_timestamp`        timestamp        NULL DEFAULT NULL,
```
[the remark this answers is not in the document]
Comment (shlomi-noach) [reply go/vt/sidecardb/schema/onlineddl/schema_migrations.sql]: Changed to `shadow_analyzed_timestamp`.

File: go/vt/vttablet/onlineddl/executor.go (hunk 2 above)
Comment (shlomi-noach) [reply go/vt/vttablet/onlineddl/executor.go]: Fixed (in two places).

File: go/vt/vttablet/onlineddl/executor.go (hunk 1 above)
Comment (shlomi-noach) [reply go/vt/vttablet/onlineddl/executor.go:1002]: Reference: https://github.com/vitessio/vitess/pull/17207

**Review by deepthi — APPROVED** [review]
